woocommerce/woocommerce
# Validate and prepare GTIN in structured data

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR removes hypens for the GTIN field and validate if is in the correct format:

As specified [here](https://schema.org/gtin) the format is:

- 8,12,13,14 length
- Only numbers



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a Product and add a valid GTIN-8 (12345678)
2. Go to the product and see the `gtin` field appears in the structured data application/ld+json in the code source
3. Create a Product and add a valid GTIN-12 (123456789012)
4. Go to the product and see the `gtin` field appears in the structured data application/ld+json in the code source
5. Create a Product and add a valid GTIN-13 (1234567890123)
6. Go to the product and see the `gtin` field appears in the structured data application/ld+json in the code source
7. Create a Product and add a valid GTIN-14 (12345678901234)
8. Go to the product and see the `gtin` field appears in the structured data application/ld+json in the code source
9. Create a Product and add a valid GTIN with hypens (1234-5678)
10. Go to the product and see the `gtin` field appears in the structured data application/ld+json in the code source as 12345678
11.  Create a Product and add a non valid GTIN (123)
12.  Go to the product and see the `gtin` field doesn't appear in the structured data application/ld+json



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
